gravitee-gateway-services/gravitee-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/DebugService.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.debug;
+
+import io.gravitee.common.component.Lifecycle;
+import io.gravitee.common.component.LifecycleComponent;
+import io.gravitee.common.service.AbstractService;
+import io.gravitee.common.util.ListReverser;
+import io.gravitee.gateway.debug.reactor.DebugReactor;
+import io.gravitee.gateway.debug.sync.DebugSyncService;
+import io.gravitee.gateway.debug.vertx.VertxDebugService;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Guillaume CUSNIEUX (guillaume.cusnieux at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class DebugService extends AbstractService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DebugService.class);
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        List<Class<? extends LifecycleComponent>> components = this.components();
+        Iterator componentsIte = components.iterator();
+
+        while (componentsIte.hasNext()) {
+            Class<? extends LifecycleComponent> componentClass = (Class) componentsIte.next();
```

Comment:
Should we simplify this loop using a for-style, instead of an interator ? like `for (Class<? extends LifecycleComponent> componentClass : components)`